packages/api-explorer/src/ApiExplorer.tsx

```diff
@@ -113,16 +113,24 @@ const ApiExplorer: FC<ApiExplorerProps> = ({
               value={{ searchSettings, setSearchSettings }}
             >
               <Page style={{ overflow: 'hidden' }}>
-                <Header
-                  specs={specs}
-                  spec={spec}
-                  specDispatch={specDispatch}
-                  toggleNavigation={toggleNavigation}
-                />
+                {envAdaptor.runAsEmbed() || (
+                  <Header
+                    specs={specs}
+                    spec={spec}
+                    specDispatch={specDispatch}
+                    toggleNavigation={toggleNavigation}
+                  />
+                )}
                 <Layout hasAside height="100%">
                   {hasNavigation && (
                     <AsideBorder pt="large" width="20rem">
-                      <SideNav api={spec.api} specKey={spec.key} />
+                      <SideNav
+                        api={spec.api}
```

Comment:
You've added logic to get the context to SideNav. Simplify things and pass  the flag in as a property as you already have it.

Again, avoid the term runAsEmbed. Suggest headless but other terms maybe appropriate.